pkg/storage/backend-storage/backend-storage.go

```diff
@@ -104,6 +108,141 @@ func PVCForMigrationTarget(pvcStore cache.Store, migration *corev1.VirtualMachin
 	return pvcForMigrationTargetFromStore(pvcStore, migration)
 }
 
+func RecoverFromBrokenMigration(client kubecli.KubevirtClient, migration *corev1.VirtualMachineInstanceMigration, pvcStore cache.Store, vmi *corev1.VirtualMachineInstance, launcherImage string) error {
+	if migration.Status.MigrationState == nil ||
+		migration.Status.MigrationState.TargetPersistentStatePVCName == migration.Status.MigrationState.SourcePersistentStatePVCName {
+		// The migration either didn't actually start, or the backend storage is RWX.
+		// In both cases we consider the migration as failed.
+		migration.Status.Phase = corev1.MigrationFailed
+		return nil
+	}
+
+	// An interrupted migration exists. Using a job to check if the source PVC contains /meta/migrated,
+	// which would indicate that the libvirt migration finished.
+	// A JobComplete condition indicates the file is present, the migration was successful and the target PVC prevails
+	// A JobFailed condition indicated the file is absent, the migration didn't finish and the source PVC prevails
+
+	jobName := "recover-" + migration.Name
+
+	job, err := client.BatchV1().Jobs(vmi.Namespace).Get(context.Background(), jobName, metav1.GetOptions{})
```

Comment:
but now, without the pull, we are going to do two get() calls for each cycle 
we should use an informer for these jobs and, if needed, push the relevant vmim back to the queue

Reply:
+1

Reply:
The second `Get()` is leftover code, will remove.
About using an informer, I'm not sure adding a jobs informer is worth the amount of code it will take and the CPU cycles it will cost...
Please keep in mind that migration recovery should be an extremely rare occurrence!

Reply:
ok, thanks! 
I'm less concerned about this at this point as we can improve after we get this fix in.